frontend/src/metabase/dashboard/actions/cards-typed.ts

```diff
@@ -239,16 +253,34 @@ export const replaceCard =
     dashboardId && trackQuestionReplaced(dashboardId);
   };
 
+export type RemoveCardFromDashboardOpts = {
+  dashcardId: DashCardId;
+  cardId: CardId | null;
+};
+
+export type RemoveCardFromDashboardAction = (
+  props: RemoveCardFromDashboardOpts,
+) => {
+  dashcardId: DashCardId;
+};
 export const removeCardFromDashboard = createThunkAction<
```

Comment:
This doesn't look like it's returning what you're expecting: 
<img width="660" alt="image" src="https://github.com/metabase/metabase/assets/1914270/12949844-a6da-48ef-aa08-a9320ca5cdca">

`createThunkAction` is not returning the value returned by the thunk/action creator (which is confusing because it's different from createAsyncThunk from rtk.
This popped out last week too with Nick, I think I'm going to make a RFC to propose not using createThunkAction anymore where possible, and prefer helpers by rtk which are way better typed

Reply:
Meanwhile let me see if I can type `createThunkAction` for the return value, as I typed the input args

Reply:
Yep, so I wasn't trying to replicate the action's actual return type but the return type we expect when we dispatch the action. Because we tend to prop-drill dashboard actions, we have a ton of places where we type the actions as their raw values (like this one)

Reply:
It could be named differently to reflect this

Reply:
> the return type we expect when we dispatch the action.

you mean what you'd end up with `const something = dispatch(actionX(params))`?
That's what I tried and the result is a Promise of what I screenshotted above

Reply:
ah! ok I see what you mean

Reply:
I'll use the `ReduxAction` in `state.ts` for now - it's not the perfect solution but it's actually accurate so that's a plus